### PR TITLE
impersonation support for the dashboard

### PR DESCRIPTION
**Pull request by mlbiam**
For #3664 added header passthrough support, test cases and docs

### CODE REVIEW

File: src/app/backend/client/manager.go [hunk 1]
```diff
@@ -50,6 +50,9 @@ const (
 	JWETokenHeader = "jweToken"
 	// Default http header for user-agent
 	DefaultUserAgent = "dashboard"
+
```
Comment (floreks) [inline src/app/backend/client/manager.go]: Remove space

File: src/app/backend/validation/validateloginstatus.go [hunk 2]
```diff
@@ -30,21 +30,35 @@ type LoginStatus struct {
 	// True if dashboard is configured to use HTTPS connection. It is required for secure
 	// data exchange during login operation.
 	HTTPSMode bool `json:"httpsMode"`
+
```
Comment (floreks) [inline src/app/backend/validation/validateloginstatus.go]: Remove space

File: src/app/frontend/chrome/userpanel/template.html [hunk 3]
```diff
@@ -18,13 +18,15 @@
   <div class="kd-auth-status kd-muted-light">
     <ng-container *ngIf="isLoginStatusInitialized">
       <ng-container [ngSwitch]="true">
-        <ng-container *ngSwitchCase="loginStatus.headerPresent"
+        <ng-container *ngSwitchCase="loginStatus.headerPresent && ! loginStatus.impersonationPresent"
```
Comment (floreks) [inline src/app/frontend/chrome/userpanel/template.html]: Remove space between `!` and `loginStatus.impersonationPresent`

File: src/app/backend/client/manager_test.go [hunk 4]
```diff
@@ -305,3 +305,301 @@ func TestClientManager_InsecureAPIExtensionsClient(t *testing.T) {
...
+				c.request, c.expectedImpersonationExtra["scope"][1], cfg.Impersonate.Extra["scope"][1])
+
+		}
+
+		if len(cfg.Impersonate.Extra["scope"]) != 2 {
+			t.Fatalf("Config(%v): Expected two scope impersonated extra but got %d",
+				c.request, len(cfg.Impersonate.Extra["scope"]))
+		}
+
+		/*
```
Comment (floreks) [inline src/app/backend/client/manager_test.go]: Unused code?

File: docs/user/README.md [hunk 5]
```diff
@@ -10,5 +10,21 @@
 * [Integrations](integrations.md)
 * [Labels](labels.md)
 
+## User Impersonation
+
+Impersonation uses a reverse proxy to inject a user's identifying information (username, groups and extra scopes) as headers in each request to the API server.  The dashboard can pass these headers to the API server if also behind a reverse proxy that can inject this information.
```
Comment (floreks) [inline docs/user/README.md]: I'd change to:

> Impersonation uses a reverse proxy to inject a user's identifying information (username, groups and extra scopes) as headers in each request to the API server.  The Dashboard can pass these headers to the API server if your reverse proxy will inject them in the requests.

File: docs/user/README.md [hunk 6]
```diff
@@ -10,5 +10,21 @@
...
+
+![Impersonation Architecture](images/dashboard-impersonation.png "Impersonation Architecture")
+
+Impersonation is useful in situations where using a user's token isn't available, such as cloud hosted Kubernetes services.  To use impersonation a reverse proxy must:
```
Comment (floreks) [inline docs/user/README.md]: s/cloud hosted/cloud-hosted

File: docs/user/README.md [hunk 7]
```diff
@@ -10,5 +10,21 @@
...
+
+1. Have a Kubernetes service account that [has RBAC permissions to impersonate other users](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#user-impersonation)
+2. Generate the `Imnpersonate-User` header with a unique name identifying the user
```
Comment (floreks) [inline docs/user/README.md]: Typo s/Imnpersonate/Impersonate

**Comment by floreks** [comment]
@mlbiam plrease amend `updates per @floreks` commit message as prow does not mention in commit messages..

**Comment by mlbiam** [comment]
@floreks done

**Pull request merged**